.github/workflows/ci.yml

```diff
@@ -0,0 +1,115 @@
+name: CI
+
+on:
+  push:
+    # Prevent duplicate runs of this workflow on our own internal PRs.
+    branches:
+      - main
+      - next/*
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+    branches:
+      - main
+      - next/*
+
+env:
+  PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: "true"
+
+jobs:
+  matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      json: ${{ steps.set_matrix.outputs.json }}
+    steps:
+      - name: Set Matrix
+        id: set_matrix
+        run: |
+          node << 'EOF'
+            const os = ["ubuntu-latest", "windows-latest", "macos-latest"];
+            const rustToolchains = ["stable", "nightly"];
+            const nodeVersions = ["14.x", "16.x", "18.x"];
+            const partialRustToolchains = ["stable"];
+            const partialNodeVersions = ["16.x"];
```

Comment:
@dherman Do you think these constants are clear enough embedded here? I originally had them in environment variables, but the logic to read and parse them added a bunch more code without doing anything but, moving them from YAML to environment variables.

Since Github Actions doesn't support array literals, they were a JSON string even when in YAML.

```yaml
env:
    RUST_TOOLCHAINS: '["stable", "nightly"]'
```